storagedriver/azure/azure.go

```diff
@@ -174,11 +193,17 @@ func (d *Driver) WriteStream(path string, offset, size int64, reader io.ReadClos
 		}
 	}
 
-	if offset != resumableOffset {
-		return storagedriver.InvalidOffsetError{Path: path, Offset: offset}
+	if offset < resumableOffset {
```

Comment:
Now that we require drivers to write to arbitrary offsets, I need to use a totally different type of blob on Azure storage (called “page blobs”), and therefore I need to implement those calls at [**MSOpenTech/azure-sdk-for-go**](https://github.com/MSOpenTech/azure-sdk-for-go/tree/master/clients/storage) and then I need to build a complex logic and tons of test cases around it. 

Unfortunately Azure Blob service does not allow range writes to arbitrary offsets. The page blob concept requires write offsets to be multiplies of 512 bytes (e.g. 0-511, 512-1023) and a single write request can be at most 4 MB (therefore I'll keep splitting the given chunk internally).

Also azure page blobs can be at most 1 TB in size. I hope there is no requirement to store that much of data in a single file? If such a requirement arises there is nothing we can do about it from Azure side.

...I'll probably need a few more days. :smile: